tor/Dockerfile

```diff
@@ -13,10 +13,39 @@ RUN \
     apk add --no-cache \
         coreutils=9.3-r1 \
         openssl=3.1.4-r1 \
-        tor=0.4.8.9-r0
+        tor=0.4.8.9-r0 \
+        go=1.20.10-r0 \
+        git=2.40.1-r0 \
+        ca-certificates=20230506-r0 \
+        libcap=2.69-r0
+
+# Install obfs4proxy transport client
+RUN git clone https://github.com/Yawning/obfs4.git /go/obfs4/
+WORKDIR /go/obfs4/
+RUN go build -o /usr/local/bin/obfs4proxy ./obfs4proxy
+
+# Install snowflake transport client
+RUN git clone --branch v2.6.1 https://gitlab.torproject.org/tpo/anti-censorship/pluggable-transports/snowflake.git /go/snowflake/
+WORKDIR /go/snowflake/client/
+RUN go get
+RUN go build -o /usr/local/bin/snowflake
+
+# Install webtunnel client
+RUN git clone https://gitlab.torproject.org/tpo/anti-censorship/pluggable-transports/webtunnel.git /go/webtunnel/
+WORKDIR /go/webtunnel/main/client/
+RUN go build -ldflags="-s -w" -o /usr/local/bin/webtunnel
+
+# Give tranposrts clients the capability to bind port 80. This line isn't necessary if
+# you use a high (unprivileged) port.
+RUN setcap \
+    'cap_net_bind_service=+ep' /usr/local/bin/obfs4proxy \
+    'cap_net_bind_service=+ep' /usr/local/bin/snowflake \
+    'cap_net_bind_service=+ep' /usr/local/bin/webtunnel 
+
+WORKDIR /
 
 HEALTHCHECK \
-    --start-period=5m \
+    --start-period=60m \
```

Comment:
Why 60! minutes?

Reply:
Connection to bridge takes up to 60 minutes. Other way Docker restarts  container every  5 minutes during bridge bootstrapping, and it will never connect to tor. 
I guess it can be user configurable or switch between two predefined values depending on the bridge configuration